app/models/concerns/audit_extensions.rb

```diff
@@ -235,10 +235,10 @@ def set_taxonomy_for(taxonomy)
   def set_taxonomies_using_associated(key_name)
     ids_arr = []
     if associated.respond_to?(:"#{key_name}_id")
-      ids_arr = [associated.send("#{key_name}_id")].compact.uniq
+      ids_arr = [associated.send("#{key_name}_id")]
     elsif associated.respond_to?(:"#{key_name}_ids")
       ids_arr = associated.send("#{key_name}_ids")
     end
-    send("#{key_name}_ids=", ids_arr)
+    send("#{key_name}_ids=", ids_arr.compact.uniq)
```

Comment:
Since we are at this, lets refactor this too? something like:

```
def set_taxonomies_using_associated(key_name)
  ids_arr = associated.respond_to?(:"#{key_name}_id") 
             ? [associated.send("#{key_name}_id")] 
             : associated.send("#{key_name}_ids")
  send("#{key_name}_ids=", ids_arr.compact.uniq)
end
```